breathecode/media/views.py

```diff
@@ -114,15 +115,17 @@ def put(self, request, media_id=None, academy_id=None):
     def delete(self, request, media_id=None, academy_id=None):
         from ..services.google_cloud import Storage
 
-        data = Media.objects.filter(
-            id=media_id, academy__id=academy_id).first()
+        data = Media.objects.filter(id=media_id).first()
         if not data:
             raise ValidationException('Media not found', code=404)
-
+        
+        if not data.academy or data.academy.id != int(academy_id):
+            raise ValidationException('You may not delete media that belongs to a different academy', 
+                slug='academy-different-than-media-academy')
+        
```

Comment:
If you remove a media, and not exist other media with the same hash, remove all the resolutions with this hash because it's orphans